File: 0125-valid-palindrome/0125-valid-palindrome.ts
function isPalindrome(s: string): boolean {
    s = Array.from(s).filter(char => /[a-zA-Z0-9]/.test(char)).join('').toLowerCase();

    const size = s.length;
    const half = Math.floor(size / 2);

    if (size % 2 === 0) {
        return s.slice(0, half) === s.slice(half).split('').reverse().join('');
    } else {
        return s.slice(0, half) === s.slice(half + 1).split('').reverse().join('');
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
# class Solution:
#     def isPalindrome(self, s: str) -> bool:
#         s = ''.join([char for char in s if char.isalnum()]).lower()
#         for i in range(len(s)//2):
#             if s[i] != s[~i]:
#                 return False
#         return True

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = ''.join([char for char in s if char.isalnum()]).lower()
        size = len(s)
        return s[:size//2] == s[size//2:][::-1] if size % 2 == 0 else s[:size//2] == s[size//2 + 1:][::-1]
